client/servicediscovery/service_discovery.go

```diff
@@ -126,14 +126,16 @@ type ServiceDiscovery interface {
 	// CheckMemberChanged immediately check if there is any membership change among the leader/followers
 	// in a quorum-based cluster or among the primary/secondaries in a primary/secondary configured cluster.
 	CheckMemberChanged() error
+	// ExecuteAndAddServingURLSwitchedCallback executes the callback once and adds it to the callback list then.
+	ExecuteAndAddServingURLSwitchedCallback(cb leaderSwitchedCallbackFunc)
```

Comment:
Why we need a `ExecuteAndAddServingURLSwitchedCallback` interface?

If we need execute some functions, I think we can execute them directly.

Reply:
`serviceDiscovery` is exposed as an interface, the caller cannot access those internal callbacks.

Reply:
Because I find it is called in only one place. So maybe `Execute` is an unnecessary abstraction.

Reply:
It will be used in the next PR also, this method is useful if we want to integrate a sub-client into the client, an immediate execution of the callback with the lastest leader URL is necessary, e.g., https://github.com/tikv/pd/pull/8939/files#diff-c2cd9790c2f82b378808c8f7326f85b46ab36aa388c95250865571df80a4679bR193.